### PR TITLE
[feat] Add frontend gql commands for AssignMates

### PR DESCRIPTION
**Pull request by SammyBaek**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #102
* #98
* __->__ #97

## Changes made
- This PR is mainly for preparation for assigning unassigned mates for frontend.
  - re-organize frontend gql files
  - add fragments for gql types
  - add assignMates and getUserData gql

## Testing
- [x] End-to-End

### CODE REVIEW

File: client/src/index.tsx [hunk 1]
```diff
@@ -7,12 +7,25 @@ import ReactDOM from "react-dom";
...
 
 import { graphqlPath } from "./api";
+import { Mate } from "./graphql/types";
 import App from "./pages/App/App";
 import reportWebVitals from "./reportWebVitals";
 
 const client = new ApolloClient({
   uri: graphqlPath,
-  cache: new InMemoryCache(),
+  cache: new InMemoryCache({
```
Comment (SammyBaek) [inline client/src/index.tsx:16]: Apollo yells at me if I don't do this

File: client/src/graphql/graphql.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    families @include(if: $withFamily) {
+      ...familyFields
+    }
+  }
+`;
+
+/**
+ * GQL Query for fetching all data
+ */
+export const GET_USER_DATA = gql`
```
Comment (SammyBaek) [inline client/src/graphql/graphql.ts:66]: This allows you to get a specified depth of info from user. Not used right now, but can be a good example for fragments and directives.

File: client/src/graphql/graphql.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  ${FAMILY_FIELDS}
+  ${MATE_FIELDS}
+  ${SYNC_FIELDS}
+`;
+
+/**
+ * GQL Query for unassigned mates and families
+ */
+export const GET_UNASSIGNED_DATA = gql`
+  query getUnassignedDataFn($withSync: Boolean = true) {
```
Comment (SammyBaek) [inline client/src/graphql/graphql.ts:87]: GQL query to get minimal info for assigning unassigned mates

File: client/src/graphql/types.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/* GraphQL related types to typescript */
```
Comment (SammyBaek) [inline client/src/graphql/types.ts:1]: This file is specifically for GQL Types to typescript

**Review by EpicPi — APPROVED** [review]
lgtm

**Pull request merged**